src/token/src/ledger.rs

```diff
@@ -74,6 +74,27 @@ impl Ledger {
         id
     }
 
+    pub fn batch_transfer(
+        &mut self,
+        from: Principal,
+        transfers: Vec<(Principal, Nat)>,
+        fee: Nat,
+    ) -> Vec<Nat> {
+        let id = self.next_id();
+
+        transfers.iter().for_each(|(to, amount)| {
```

Comment:
I really wonder about using functional style instead of `for` loops. In my opinion it makes the code less readable, but it's just an opinion. @hagsteel , do you have an opinion on this?

Reply:
Do I ever!

Historically `for_each` has been "known" to be faster than regular `for` loops:
This is possibly misleading since it might be related to bounds checking etc. and this was something you would see a few years ago in Rust, but not so much today.

If we were to look at the asm for a for loop and a `for_each` doing something basic, it would most likely be the same (or at least very close to the same, zero cost abstractions and all).

The long and short of it is: there is no significant difference. It's a matter of taste.
In this instance I think it's fine since it isn't doing anything complicated with 100 iterator adapters.

The nice thing about regular for-loops is that it's something we all know from all other imperative programming languages we have come across in our lives, and Rust even translates
`for x in &y {}` to `for x in y.iter() {}`.

However `for_each` has it's place, given that we can call it with anything that implements the `FnMut` trait as an argument, so we can write code like this:
`y.into_iter().for_each(print_it);`

Conclusion: use whatever works for you. Iterator adapters are super nice, but don't combine 200 of them expecting anyone to read it. In this case there is no confusion as it's short